### ISSUE TITLE
Ustream.tv extractor fails with HTTP Error 401: Unauthorized in 2014.10.18

### ISSUE THREAD

**Issue by microace** [rg3/youtube-dl#3998]
youtube-dl --verbose http://www.ustream.tv/recorded/54195525
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['--verbose', 'http://www.ustream.tv/recorded/54195525']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2014.10.18
[debug] Git HEAD: b100194
[debug] Python version 3.4.2 - Darwin-14.0.0-x86_64-i386-64bit
[debug] Proxy map: {}
[ustream] 54195525: Downloading webpage
[ustream] 54195525: Extracting information
[debug] Invoking downloader on 'http://tcdn.ustream.tv/video/54195525'
ERROR: unable to download video data: HTTP Error 401: Unauthorized
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/site-packages/youtube_dl/YoutubeDL.py", line 1037, in process_info
    success = dl(filename, info_dict)
  File "/usr/local/lib/python3.4/site-packages/youtube_dl/YoutubeDL.py", line 1013, in dl
    return fd.download(name, info)
  File "/usr/local/lib/python3.4/site-packages/youtube_dl/downloader/common.py", line 292, in download
    return self.real_download(filename, info_dict)
  File "/usr/local/lib/python3.4/site-packages/youtube_dl/downloader/http.py", line 64, in real_download
    data = self.ydl.urlopen(request)
  File "/usr/local/lib/python3.4/site-packages/youtube_dl/YoutubeDL.py", line 1266, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 461, in open
    response = meth(req, response)
  File "/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 571, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 499, in error
    return self._call_chain(*args)
  File "/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 433, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 579, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 401: Unauthorized

**Comment by P2K9** [comment]
Since this is still open I'll throw mine in there as well. Trying to rip from ustream and get the same error.

youtube-dl --verbose "http://www.ustream.tv/recorded/57082787"
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['--verbose', 'http://www.ustream.tv/recorded/5708278
7']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2014.12.17.2
[debug] Python version 2.7.8 - Windows-7-6.1.7601-SP1
[debug] exe versions: none
[debug] Proxy map: {}
[ustream] 57082787: Downloading webpage
[ustream] 57082787: Extracting information
[debug] Invoking downloader on u'http://tcdn.ustream.tv/video/57082787'
ERROR: unable to download video data: HTTP Error 401: Unauthorized
Traceback (most recent call last):
  File "youtube_dl\YoutubeDL.pyo", line 1136, in process_info
  File "youtube_dl\YoutubeDL.pyo", line 1112, in dl
  File "youtube_dl\downloader\common.pyo", line 297, in download
  File "youtube_dl\downloader\http.pyo", line 67, in real_download
  File "youtube_dl\YoutubeDL.pyo", line 1370, in urlopen
  File "urllib2.pyo", line 410, in open
  File "urllib2.pyo", line 523, in http_response
  File "urllib2.pyo", line 448, in error
  File "urllib2.pyo", line 382, in _call_chain
  File "urllib2.pyo", line 531, in http_error_default
HTTPError: HTTP Error 401: Unauthorized

**Comment by cadesalaberry** [comment]
Here is mine in case it turns out to be useful:
```bash
$ youtube-dl --verbose  http://www.ustream.tv/channel/nomadlive-tv
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['--verbose', 'http://www.ustream.tv/channel/nomadlive-tv']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2014.10.30
[debug] Python version 2.7.6 - Linux-3.13.0-45-generic-x86_64-with-Ubuntu-14.04-trusty
[debug] exe versions: avconv 9.16-6, avprobe 9.16-6
[debug] Proxy map: {}
[ustream:channel] nomadlive-tv: Downloading webpage
[ustream:channel] nomadlive-tv: Downloading video information (next: 1)
[download] Downloading playlist: 18155672
[ustream:channel] playlist 18155672: Collected 5 video ids (downloading 5 of them)
[download] Downloading video #1 of 5
[ustream] 56226348: Downloading webpage
[ustream] 56226348: Extracting information
[debug] Invoking downloader on u'http://tcdn.ustream.tv/video/56226348'
ERROR: unable to download video data: HTTP Error 401: Unauthorized
Traceback (most recent call last):
  File "/home/username/.local/lib/python2.7/site-packages/youtube_dl/YoutubeDL.py", line 1052, in process_info
    success = dl(filename, info_dict)
  File "/home/username/.local/lib/python2.7/site-packages/youtube_dl/YoutubeDL.py", line 1028, in dl
    return fd.download(name, info)
  File "/home/username/.local/lib/python2.7/site-packages/youtube_dl/downloader/common.py", line 292, in download
    return self.real_download(filename, info_dict)
  File "/home/username/.local/lib/python2.7/site-packages/youtube_dl/downloader/http.py", line 64, in real_download
    data = self.ydl.urlopen(request)
  File "/home/username/.local/lib/python2.7/site-packages/youtube_dl/YoutubeDL.py", line 1286, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python2.7/urllib2.py", line 410, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 523, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 448, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 531, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
HTTPError: HTTP Error 401: Unauthorized
```

**Comment by mlindner** [comment]
When is this going to get fixed? This has been broken for a long time.

**Comment by dstftw** [comment]
When AMF will be supported.

**Comment by mlindner** [comment]
Is this really necessary? I can pull the streaming FLV URL out of chrome developer tools network tab.

**Comment by dstftw** [comment]
Yes, ustream player received this data via AMF and then fetches flv with HTTP GET. That's why you can see it in developer tools.
Feel free to suggest alternative method if you know one.

**Comment by yan12125** [comment]
I've implement an alternative approach to fetch the video URL. youtube-dl should works for ustream.tv in the next version.

**Comment by dstftw** [comment]
This has been fixed in youtube-dl 2015.04.26. See our [FAQ](https://yt-dl.org/update) if you have problems updating.
